core/directory.go

```diff
@@ -282,6 +282,17 @@ func (dir *Directory) Diff(ctx context.Context, other *Directory) (*Directory, e
 	return NewDirectory(ctx, llb.Diff(st, otherSt), rel, platform)
 }
 
+func (dir *Directory) Without(ctx context.Context, path string) (*Directory, error) {
+	st, cwd, platform, err := dir.Decode()
+	if err != nil {
+		return nil, err
+	}
+
+	st = st.File(llb.Rm(path))
```

Comment:
huge nit: I almost wonder if it makes sense to default to `AllowNotFound: true` here. Semantically, that seems in line with `Without`. If you disagree though it's not a blocker at all.

Also either way, probably good to have a follow up issue for `AllowNotFound` (or the inverse depending on our default) and `AllowWildcard` (I think at least one of those was specifically requested by users in Europa, so I'm sure it will come up again soon here)

Reply:
Thought about that but it felt safer to default `AllowNotFound: false` because otherwise someone might not notice that a large artifact they're trying to remove has changed locations and is no longer being removed. I think it's safer to let the user know something they attempted to do ended up being a no-op.

+1 to supporting configuring both though. Opened https://github.com/dagger/dagger/issues/3338